chore: Clean up Xcode config

klochowicz opened the pull request:
- declary sentry as FirezoneKit dependency
- add config for non-Xcode LSP config 
- add some more info in README

klochowicz committed:
build: Keep sentry inside FirezoneKit

There's no need to have it as a project-wide dependency, it can be
encapsulated inside FirezoneKit

klochowicz committed:
chore(macos): Generate LSP information for non-XCode IDEs

Useful for sourcekit-lsp (working with Swift code within neovim, zed,
emacs etc)

klochowicz committed:
docs: Update Apple building instructions

Be a bit more specific that Xcode builds `connlib`
(its project file is hard to read)

klochowicz committed:
chore: Simplify Apple build instructions

No need to manually build connlib

klochowicz committed:
fix: Use 'exact' package qualified for Swift dependencies

This pins the package to exact version.
Dependabot needs this to work properly with swift.

jamilbk commented on .github/dependabot.yml:
```diff
@@ -136,12 +136,14 @@ updates:
           - com.google.firebase:firebase-crashlytics-ndk
           - com.google.firebase:firebase-analytics-ktx
 
+# FIXME: Bring back regular swift defaults (weekly, 28 day cooldown)
```
You shouldn't need this as any changes to dependabot's config will cause an immediate check.

You can also view dependabot runs in the "Insights" tab of the repo.

klochowicz replied:
good to know about the insights tab!

However, not sure that we'd be able to spot:
- if  I revert this temporary change, then the dependabot config will not trigger an immediate check
- that's still OK, because I believe I can manually trigger a check
- however, apparently there's no option to see the updates waiting for the cooldown period
- due to our 28 day cooldown, we'll likely not going to see whether it works for macOS (as I fear  we're only one version behind with sentry, and the latest one is younger than the cooldown period) 

Happy to just wait it out, in the end it's just a nice side effect of these changes that dependabot should start working properly. Let me know what you prefer @jamilbk .

klochowicz replied:
update: I've just downgraded to 8.49 (the version used earlier). this means that dependabot insight should follow straight away.

jamilbk reviewed (commented):
Maybe mention that you also need to launch Xcode to get profiles set up, use its debugger, and monitor memory / energy usage meters?

In other words, editing source code is just one aspect of macOS / iOS development - which devs are free to use whatever they like.

But for the entire development workflow it's probably good to encourage Xcode (and Instruments) usage?

For example, iOS has a 50 MB hard cap on memory usage in the network extension. Whenever we make changes to our threading model it's a good idea to double-check we don't go over there.

klochowicz committed:
docs: Make it clear that Xcode is fundamental on macOS

Elaborate that only code editing can be done outside of Xcode and
describe what still needs to be configured/used from it.

Advertise Instruments for people new to macOS/iOS development.

klochowicz committed:
chore(macos): Set sentry dependency to 8.49

Keep the version that was set before; dependabot will schedule an update.

jamilbk reviewed (approved):
LGTM, the markdown I believe just needs a `prettier` run on it.

klochowicz committed:
chore(docs): Reformat with prettier
